pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -446,10 +485,19 @@ func (nc *Controller) doNoScheduleTaintingPassWorker() {
 			return
 		}
 		nodeName := obj.(string)
-
+		needsToRequeue := false
 		if err := nc.doNoScheduleTaintingPass(nodeName); err != nil {
-			// TODO (k82cn): Add nodeName back to the queue.
 			klog.Errorf("Failed to taint NoSchedule on node <%s>, requeue it: %v", nodeName, err)
+			needsToRequeue = true
+		}
+		if err := nc.reconcileNodeLables(nodeName); err != nil {
```

Comment:
It's better to add a TODO here, e.g. we'll remove this after 1.18.

Reply:
We still need this in 1.18 (see the KEP). There are also other labels under discussion that are may rely this mechanism. I'll add a TODO to re-evaluate in 1.19

Reply:
ok to me :)